### PR TITLE
test(compiler-vapor): add more test for expression parsing (v-on, v-slot, v-for)

### PR DESCRIPTION
**Pull request by Simon-He95**

### CODE REVIEW

File: packages/compiler-vapor/__tests__/compile.spec.ts [hunk 1]
```diff
@@ -193,7 +193,49 @@ describe('compile', () => {
...
+    })
+
+    test('v-slot', () => {
+      const code = compile(
+        `
+    <script setup lang="ts">
+    import { Foo, Bar, Baz, Qux, Fred } from './x'
+    const a = 1
+    function b() {}
+    </script>
```
Comment (LittleSound) [inline packages/compiler-vapor/__tests__/compile.spec.ts:223]: The compile function in compiler-vapor only processes the code in the `<template>`, and `<script setup lang="ts">` in the test case may be beyond its scope.

Comment (LittleSound) [inline packages/compiler-vapor/__tests__/compile.spec.ts:223]: The compile function in compiler-vapor only processes the code in the `<template>`, and `<script setup lang="ts">` in the test case may be beyond its scope.

Comment (Simon-He95) [reply packages/compiler-vapor/__tests__/compile.spec.ts]: ok~

**Review by sxzz — CHANGES_REQUESTED** [review]
`compile.spec.ts` no longer accepts additional tests. Our goal is to align with the existing tests (runtime-core, compiler-core).

See https://github.com/vuejs/core-vapor/commit/6608bb31973d35973428cae4fbd62026db068365

**Pull request closed without merging**